compiler/damlc/lib/DA/Cli/Damlc/DependencyDb.hs

```diff
@@ -246,24 +279,143 @@ validatePkgId pkgId = do
         fail $ "Invalid package ID dependency in daml.yaml: " <> pkgId
     pure $ LF.PackageId $ T.pack pkgId
 
+-- Package resolution
+---------------------
+
+data FullPkgName = FullPkgName
+    { pkgName :: LF.PackageName
+    , pkgVersion :: LF.PackageVersion
```

Comment:
```suggestion
    { pkgName :: !LF.PackageName
    , pkgVersion :: !LF.PackageVersion
```

Reply:
Probably want this on everything here, I don’t think any of these should be lazy.